### PR TITLE
isl: update to 0.23 (with cloog and soname)

### PR DESCRIPTION
**Pull request by dkwo**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by dkwo** [comment]
See also #24424
This time I only revbump cloog.

File: srcpkgs/isl/template [hunk 1]
```diff
@@ -23,8 +24,9 @@ post_install() {
 }
 
 isl-devel_package() {
-	depends="gmp-devel isl>=${version}_${revision}"
+ 	depends="${makedepends} ${sourcepkg}>=${version}_${revision}"
 	short_desc+=" - development files"
+ 	conflicts="isl-devel>=0"
```
Comment (tornaria) [inline srcpkgs/isl/template:29]: Why conflict with itself?

Comment (dkwo) [reply srcpkgs/isl/template]: Thanks for comments both here and in the other pr.
I edited accordingly.

**Comment by dkwo** [comment]
@pullmoll @tornaria how does it look now?

File: srcpkgs/isl/template [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
 
 post_install() {
+	vlicense LICENSE
 	# Remove gdb python files.
 	rm -f ${DESTDIR}/usr/lib/*.py
 }
 
 isl-devel_package() {
-	depends="gmp-devel isl>=${version}_${revision}"
+ 	depends="${makedepends} ${sourcepkg}>=${version}_${revision}"
```
Comment (ericonr) [inline srcpkgs/isl/template:28]: You added an extra space here.

**Review by ericonr — COMMENTED** [review]
You're updating two packages here, you should have a commit for each.

**Comment by dkwo** [comment]
@ericonr Thanks, I think I fixed it.

**Comment by tornaria** [comment]
Looks good to me, I compiled sagemath-9.2 using this without any issue.

File: srcpkgs/isl/template [hunk 3]
```diff
@@ -1,29 +1,31 @@
 # Template file for 'isl'
 pkgname=isl
-version=0.12.2
-revision=2
+version=0.23
+revision=1
+wrksrc="isl-${version}"
```
Comment (ericonr) [inline srcpkgs/isl/template:5]: this is default, you don't need to specify wrksrc.

**Review by ericonr — COMMENTED** [review]
The isl commit message should mention that it's fixing the license and homepage.

File: srcpkgs/cloog/template [hunk 4]
```diff
@@ -10,8 +10,8 @@ short_desc="Library that generates loops for scanning polyhedra"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="LGPL-2.1-or-later"
 homepage="http://www.bastoul.net/cloog/"
-distfiles="http://www.bastoul.net/cloog/pages/download/cloog-$version.tar.gz"
-checksum=325adf3710ce2229b7eeb9e84d3b539556d093ae860027185e7af8a8b00a750e
+distfiles="https://github.com/periscop/cloog/releases/download/cloog-${version}/cloog-${version}.tar.gz"
```
Comment (ericonr) [inline srcpkgs/cloog/template:13]: What info do we have on this being an "official" fork?

Comment (dkwo) [reply srcpkgs/cloog/template:13]: The contributors include the original author and most of the people who are also active with isl and cloog development and on cloog-development googlegroup.

**Comment by dkwo** [comment]
@ericonr is the `cloog` repo fine or should I revert it back to the previous (obsolete) one?

**Comment by ericonr** [comment]
Ok, I think the cloog commit should mention that it's changing the distfiles origin, but otherwise LGTM. Just fix the commit message and we are good.

**Comment by dkwo** [comment]
@ericonr Done.

**Pull request merged**